enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,256 @@
+we need a way to insall such cluster without extra node dependency for bootstrap.
+
+This enhancement describes the flow for installing Single Node Openshift using liveCD that perform the bootstrap logic and later reboot to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Removing the bootstrap node is a key for the 5G RAN use case and to the more general topic of single-node OCP clusters for edge use cases.
+
+Requiring the axillary node for installing a Single Node Openshift brings to play some requirements:
+1. The obvious additional node, in most cases there isn't additional Hardware/VM at the edge site.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node Openshift
+
+### Goals
+
+* This enhancement describes an approach for insallting Single Node OpenShift 
+  for production use.
+* Minimal changes to openshift installer, this implementation shouldn't effect existing deployment flows.
+* Installation should result a clean Single Node Openshift, without bootstrap leftovers.
+* Rely on a livecd, so it would be able to trigger the full installation using virtual media.
+* Self-Contained bootstrap, The bootstrap flow is expressed as a single Ignition config which is layered on top of a pristine RHCOS image. It's important that we limit our use of this pattern to just the Ignition configs - as opposed to including other assets directly in the ISO image. 
+
+### Non-Goals
+
+* This enhancement does not address similar insallation flow for 3 nodes cluster.
+* This enhancement does not address high-availability for single-node
+  deployments.
+* This enhancement does not address single-node-developer cluster-profile installation.
+
+## Proposal
+
+Execute the bootstrap flow on a liveCD.
+Enrich the master ignition with the control plane static pods and write this ignition.
+Write the updated master ignition + RHCOS to disk and reboot.
+Post reboot the node will complete the installation.
+
+### User Stories
+
+#### As a user, I can deploy OpenShift in a supported single-node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+ignition configuration.
+The user will boot RHCOS liveCD with this ignition to create a single-node deployment.
+
+### Implementation Details/Notes/Constraints
+
+The openshift installer `create ignition-configs` command will generate a `bootstrap-in-place.ign`
+file, when the number of replicas for the control plane (in the `install-config.yaml`) is `1`.
+
+The `bootstrap-in-place.ign` will be embedded into an RHCOS liveCD using coreos-install embed command.
```

Comment:
are u sure? Maybe it can be served? Or Assisted Installer

Reply:
Right, I think this Ignition will actually end up being embedded inside a larger one that includes automation to run `coreos-installer` and potentially other things.  

The simple example I give is that one needs to specify at least which physical drives to use for `coreos-installer`; that is a user choice.

Reply:
@romfreiman, you are correct it can be served (I'll add it as an option).
I'm pretty sure assisted installer can provide it as well (as it does today with the bootstrap ignition).